executor/adapter.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2015 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/column"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/field"
+	"github.com/pingcap/tidb/infoschema"
+	"github.com/pingcap/tidb/optimizer/plan"
+	oplan "github.com/pingcap/tidb/plan"
+	"github.com/pingcap/tidb/rset"
+	"github.com/pingcap/tidb/util/format"
+	"github.com/pingcap/tidb/util/types"
+)
+
+// adapter wraps a executor, implements rset.Recordset interface
+type recordsetAdapter struct {
```

Comment:
The `rset.Recordset` is an old interface, but is widely used, implement this interface makes the new plan transition smooth.